azure-pipelines.yml

```diff
@@ -107,17 +107,17 @@ jobs:
           choco upgrade llvm
 
           ## To list all installed packages:
-          # choco list --localonly
+          choco list
 
           # Remove other gcc compilers
 
           ## This would be the best way to uninstall, but is SLOW:
           # choco uninstall llvm mingw rtools strawberryperl -x
 
           # Remove mingw gcc
-          rm /c/ProgramData/Chocolatey/bin/gcc
-          rm /c/ProgramData/Chocolatey/bin/g++
-          rm /c/ProgramData/Chocolatey/bin/c++
+          #rm /c/ProgramData/Chocolatey/bin/gcc
+          #rm /c/ProgramData/Chocolatey/bin/g++
+          #rm /c/ProgramData/Chocolatey/bin/c++
```

Comment:
Thanks for fixing this! I don't really understand the context and possible ramifications around this so I am hesitant to approve (and forget). Do you know understand why this script made a point in removing the other compilers and why its no longer necessary?

Reply:
I don't know what was installed before because the package listing was commented out:
```
          ## To list all installed packages:
          # choco list --localonly
```

When I changed that to (`--localonly` didn't work anymore)
```
          ## To list all installed packages:
          choco list
```
I saw
```
+ choco list
Chocolatey v2.2.2
7zip.install 23.1.0
ant 1.10.14
cmake.install 3.27.6
composer 6.3.0
docker-compose 1.29.2
DotNet3.5 3.5.20160716
gradle 8.3.0
hg 5.0.0
imagemagick 7.1.1.1900
imagemagick.app 7.1.1.1900
InnoSetup 6.2.2
jq 1.7.0
julia 1.9.3
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
kubernetes-cli 1.28.2
kubernetes-helm 3.12.3
llvm 17.0.2
maven 3.8.7
Minikube 1.31.2
nginx 1.25.2
nodejs 18.18.0
nodejs.install 18.18.0
nsis 3.8.0
nsis.install 3.8.0
nssm 2.24.101.20180116
nuget.commandline 6.7.0
packer 1.9.4
php 8.2.11
pulumi 3.86.0
R.Project 4.3.1
rtools 4.3.5550
sbt 1.9.6
strawberryperl 5.32.1.1
swig 4.1.1
tortoisesvn 1.14.5.29465
vcredist140 14.36.32532
vswhere 3.1.7
wixtoolset 3.11.2
52 packages installed.
```

The comments in the code were
``` 
         ## This would be the best way to uninstall, but is SLOW:
          # choco uninstall llvm mingw rtools strawberryperl -x

          # Remove mingw gcc
          rm /c/ProgramData/Chocolatey/bin/gcc
          rm /c/ProgramData/Chocolatey/bin/g++
          rm /c/ProgramData/Chocolatey/bin/c++
```
I didn't see `mingw` in the list of installed Chocolatey packages, but I did see `llvm`, `rtools`, and `strawberryperl`. I also didn't see any other Chocolatey packages listed that looked like they would provide gnu compilers. I suspect that they used to include mingw, but now they don't. But I am just guessing.

Reply:
rtools presumably provides mingw gcc. As does strawberrypearl. I added this because these compiler were picked up before llvm clang. If clang gets picks up first, then great.

The package listing isn't necessary, it's just there for when we want to debug.

Reply:
If we don't list things when it works, it is hard to know what changed when it doesn't. So I would like to leave the listing. It would have been helpful to have while debugging this issue.

Reply:
@stefanv Do you know where rtools or strawberrypearl install `gcc`? Because it isn't in `/c/ProgramData/Chocolatey/bin`.